vault/router.go

```diff
@@ -250,8 +250,37 @@ func (r *Router) routeCommon(req *logical.Request, existenceCheck bool) (*logica
 
 	// Reset the request before returning
 	defer func() {
-		req.Path = original
+		// We only run this if resp is not nil, so for instance we don't during
+		// an existence check
+		if resp != nil {
```

Comment:
This logic does not feel like it should go in here. I would move this out to the request handling logic instead of the router.